jenkins-scripts/docker/lib/docker_generate_dockerfile.bash

```diff
@@ -33,24 +33,32 @@ export APT_PARAMS=
 
 GZDEV_DIR=${WORKSPACE}/gzdev
 GZDEV_BRANCH=${GZDEV_BRANCH:-master}
+if [[ "$ghprbSourceBranch" =~ matching_branch\/ ]]; then
+  GZDEV_TRY_BRANCH=$ghprbSourceBranch
+fi
 
 dockerfile_install_gzdev_repos()
 {
 cat >> Dockerfile << DELIM_OSRF_REPO_GIT
 RUN rm -fr ${GZDEV_DIR}
-RUN git clone --depth 1 https://github.com/ignition-tooling/gzdev -b ${GZDEV_BRANCH} ${GZDEV_DIR}
+RUN git clone https://github.com/ignition-tooling/gzdev -b ${GZDEV_BRANCH} ${GZDEV_DIR}
+RUN bash -c "if [[ -n \"$GZDEV_TRY_BRANCH\" ]]; then git -C ${GZDEV_DIR} fetch origin \"$GZDEV_TRY_BRANCH\"; fi || true"
+RUN bash -c "if [[ -n \"$GZDEV_TRY_BRANCH\" ]]; then git -C ${GZDEV_DIR} checkout \"$GZDEV_TRY_BRANCH\"; fi || true"
```

Comment:
Could we merge this git command with the previous to save the duplication of the if?

Reply:
merged in https://github.com/ignition-tooling/release-tools/pull/572/commits/2c6faf12af27a5caf69e208d9eceb3dfe1382e5b